### ISSUE TITLE
[Feature Request] number input formatting

### ISSUE THREAD

**Issue by mmahalwy** [mantinedev/mantine#763]
### Usage example, including component/hook API

Similar to the decimal separator, would be great to have a format for the number. This is helpful for inputs that are for currency, percentages, etc. 

### Possible implementation – describe how the feature can be implemented

```ts
<NumberInput
  formatter={value => `$ ${value}`}
  value={100}
/>
```

### Do you want to contribute this feature and create a pull request

Yes

**Comment by rtivital** [comment]
This is a great feature, let's add it, feel free to submit a PR

**Comment by jerebtw** [comment]
Would be good then if you also add a `parse` function. (To transform the current value back to a number)

**Comment by mmahalwy** [comment]
I'll create a PR for this tomorrow. It's actually a blocker for us ATM so it's high on our list

**Comment by mmahalwy** [comment]
@rtivital @jerebtw have a POC PR here: https://github.com/mantinedev/mantine/pull/774

PTAL and would love your feedback

**Comment by rtivital** [comment]
Released in 3.6.5
